libbeat/feature/details.go

```diff
@@ -0,0 +1,79 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package feature
+
+import "fmt"
+
+// Describable contains general information for a specific feature, the fields will be used to report
+// useful information by the factories or any future CLI.
+type Describable interface {
+	// Stability is the stability of the Feature, this allow the user to filter embedded functionality
+	// by their maturity at runtime.
+	// Example: Beta, Experimental, Stable or Undefined.
+	Stability() Stability
+
+	// Doc is a one liner describing the current feature.
+	// Example: Dissect allows to define patterns to extract useful information from a string.
+	Doc() string
+
+	// FullName is the human readable name of the feature.
+	// Example: Jolokia Discovery
+	FullName() string
+
+	String() string
```

Comment:
What do we need `String` for? Is it supposed to be the short name?

Reply:
I've removed but details will still implement the stringer interface.